File: dotWine/drive_c/XEClient/odp.net/samples/Globalization/src/ConnectionManager.cs
/****************************************************************************
* @author :  Chandar
* @version                       :  1.0
* Development Environment        :  Microsoft Visual Studio .Net 1.0
* Name of the File               :  ConectionManager.cs
* Creation/Modification History  :
*                                   12-Oct-2002     Created 
*
* Overview:
* This class manages the database connection used by the sample. 
* It defines methods to create and close the database connection.

***************************************************************************/
//add the namespaces used by program
using System;
using System.Windows.Forms;
using Oracle.DataAccess.Client;

namespace OracleGlobalizationSample  //namespace for sample
{
	/// <summary>
	/// Summary description for ConnectionManager.
	/// </summary>
	public class ConnectionManager
	{
		public static OracleConnection conn;
		public ConnectionManager()
		{}
		
		/*******************************************************************
		* The purpose of this method is to get the database connection 
		* using the connection details from ConnectionParams.cs
		* Note: Replace the datasource parameter with your datasource value
		* in ConnectionParams.cs file.
		********************************************************************/
		public static bool getDBConnection()
		{
			try
			{
				//Connection Information	
				string connectionString = 
					//username
					"User Id=" + Config.Username +

					//password
					";Password=" + Config.Password +

					//datasource  
					";Data Source=" + Config.Datasource;

					
				//create Connection object using the above connect string
				conn = new OracleConnection(connectionString);

				//Open database connection
				conn.Open();
				return true;
			}
				// catch exceptions while connecting to the database 
			catch (Exception ex) 
			{
				//Display error message
				MessageBox.Show(ex.ToString());
				return false;
			}
		}
        
		/*******************************************************************
		* The purpose of this method is to close the database connection 
		********************************************************************/
		public static void closeConnection()
		{
			conn.Close();
		}

	}
}

File: dotWine/drive_c/XEClient/odp.net/samples/LOB/UpdLob/src/ConnectionManager.vb
'**************************************************************************
'@author                        :  Jagriti
'@version                       :  1.0
'Development Environment        :  MS Visual Studio.NET
'Name of the File               :  ConnectionManager.vb
'Creation/Modification History  :
'                                 18-Feb-2003     Created
'Overview:
'This class manages the database connection used by the sample. 
'It defines methods to create and close the database connection.
'*************************************************************************

Imports System
Imports Oracle.DataAccess.Client

Public Class ConnectionManager
    Public Shared conn As OracleConnection

    '*******************************************************************
    ' The purpose of this method is to get the database connection 
    ' using the connection details from ConnectionParams.vb
    ' Note: Replace the datasource parameter with your datasource value
    ' in ConnectionParams.vb file.
    '********************************************************************/
    Public Shared Function getDBConnection() As Boolean
        Try
            ' Setting database parameters
            ConnectionParams.setparams()

            ' Connection Information	
            Dim connectionString As String = "Data Source=" + ConnectionParams.datasource & _
                                             ";User ID=" + ConnectionParams.username & _
                                             ";Password=" + ConnectionParams.password & _
                                             ";Min Pool Size=" + ConnectionParams.minpoolsize & _
                                             ";Incr Pool Size=" + ConnectionParams.incrpoolsize & _
                                             ";Decr pool Size=" + ConnectionParams.decrpoolsize

            ' Create Connection object using the above connect string
            conn = New OracleConnection(connectionString)

            ' Open database connection
            conn.Open()
            Return True

            ' catch exceptions while connecting to the database 
        Catch ex As Exception
            Return False

        End Try
    End Function

    '*******************************************************************
    ' The purpose of this method is to close the database connection 
    '********************************************************************
    Public Shared Sub closeConnection()
        conn.Close()
        conn.Dispose()
    End Sub

End Class
